server/admin_management/api/app/schemas.py

```diff
@@ -96,3 +152,41 @@ class FamAppAdminGet(BaseModel):
     application: FamApplicationBase
 
     model_config = ConfigDict(from_attributes=True)
+
+
+# -------------------------- FAM Access Control Privilege (Delegated Admin) -------------------------- #
+class FamAccessControlPrivilegeCreateRequest(BaseModel):
+    user_name: Annotated[str, StringConstraints(min_length=3, max_length=20)]
+    user_type_code: famConstants.UserType
+    role_id: int
+    forest_client_number: Union[
+        List[Annotated[str, StringConstraints(min_length=1, max_length=8)]], None
+    ] = None
+
+    model_config = ConfigDict(from_attributes=True)
+
+
+class FamAccessControlPrivilegeCreate(BaseModel):
+    user_id: int
+    create_user: Annotated[str, StringConstraints(max_length=60)]
+    role_id: int
+
+    model_config = ConfigDict(from_attributes=True)
+
+
+class FamAccessControlPrivilegeGet(BaseModel):
+    access_control_privilege_id: int
+    user_id: int
+    role_id: int
+    user: FamUserInfo
+    role: FamRoleWithClient
+
+    model_config = ConfigDict(from_attributes=True)
+
+
+class FamAccessControlPrivilegeCreateResponse(BaseModel):
+    status_code: int
+    detail: FamAccessControlPrivilegeGet
+    error_message: Optional[str] = None
+
+    model_config = ConfigDict(from_attributes=True)
```

Comment:
Hi Catherine, this `FamAccessControlPrivilegeCreateResponse` seems to be different from normal schema.py objects.
Questions are:
- What is this `status_code`? Is it our internal status code and different than Http Status Code so we need to defined a property for it?
- Why is it to have success response model return from endpoint but having `error_message` when this seems to indicate that it should be an error response? For example, in case of success creation of a resource in backend, it should have (either status code 201 or 200, like what indicated from FastAPI):
![image](https://github.com/bcgov/nr-forests-access-management/assets/81595625/5eb69a60-674f-47ee-b9c3-f1ff22fc315b)
In case of error(s) happens in backend, depending on what is failing, normally it would be "403", "404", "409" or "500". FastAPI has indicated some error handling in documentation as well.
 
- Do we need this `model_config`? By looking at the properties, maybe it does not qualify for this `model_config` anymore?

We can discuss what' this class purpose is. Thanks Catherine!

Reply:
Hmm... Just saw the service code... I am getting a bit understanding what you are trying to do now... I think I remembered you mentioned before you are doing (this time) a bulk creation .... Let me understand a bit more on your code. Thanks!